apps/authorization/tests/test_data_access_grant.py

```diff
@@ -169,72 +184,81 @@ def test_thirteen_month_app_type_with_switch_limit_data_access(self):
 
     def test_creation_on_approval(self):
-        redirect_uri = 'http://localhost'
-        # create a user
-        user = self._create_user('anna', '123456')
-        capability_a = self._create_capability('Capability A', [])
-        capability_b = self._create_capability('Capability B', [])
-        # create an application and add capabilities
-        application = self._create_application(
-            'an app',
-            grant_type=Application.GRANT_AUTHORIZATION_CODE,
-            redirect_uris=redirect_uri)
-        application.scope.add(capability_a, capability_b)
+        user = self._create_user("anna", '123456')
+        application, fhir_id = self.setup_test_application_with_user(user)
+        # errors if DNE or more than one is found
+        DataAccessGrant.objects.get(beneficiary=user.id, application=application.id)
 
-        # user logs in
-        request = HttpRequest()
-        self.client.login(request=request, username='anna', password='123456')
+    @override_switch('expire_grant_endpoint', active=True)
+    def test_delete_authenticated_user_grant(self):
+        # 1. create user, app, ac
+        user, application, ac = self._create_user_app_token_grant(
+            first_name="first",
+            last_name="last1",
+            fhir_id="-20140000008325",
+            app_name="test_app1",
+            app_username="devuser1",
+            app_user_organization="org1",
+        )
 
-        payload = {
-            'client_id': application.client_id,
-            'response_type': 'code',
-            'redirect_uri': redirect_uri,
-        }
-        response = self.client.get('/v1/o/authorize', data=payload)
-        # post the authorization form with only one scope selected
-        payload = {
-            'client_id': application.client_id,
-            'response_type': 'code',
-            'redirect_uri': redirect_uri,
-            'scope': ['capability-a'],
-            'expires_in': 86400,
-            'allow': True,
-        }
-        response = self.client.post(response['Location'], data=payload)
+        user, application_2, ac = self._create_user_app_token_grant(
+            first_name="first",
+            last_name="last1",
+            fhir_id="-20140000008325",
+            app_name="test_app2",
+            app_username="devuser2",
+            app_user_organization="org2",
+        )
 
-        self.assertEqual(response.status_code, 302)
-        # now extract the authorization code and use it to request an access_token
-        query_dict = parse_qs(urlparse(response['Location']).query)
-        authorization_code = query_dict.pop('code')
-        token_request_data = {
-            'grant_type': 'authorization_code',
-            'code': authorization_code,
-            'redirect_uri': redirect_uri,
-            'client_id': application.client_id,
-        }
-        response = self.client.post(reverse('oauth2_provider:token'), data=token_request_data)
+        self.client.logout()
```

Comment:
This line can be removed, since there is a `.logout()` at # 3 on line 2019.

Reply:
Good catch!  I was messing around with logout at different spots and I missed that one.